lib/OptimizationMOI/src/OptimizationMOI.jl

```diff
@@ -155,6 +162,9 @@ function MOI.eval_hessian_lagrangian(
     return
 end
 
+MOI.objective_expr(prob::MOIOptimizationProblem) = prob.f.expr
+MOI.constraint_expr(prob::MOIOptimizationProblem,i) = prob.f.cons_expr
```

Comment:
This needs to return the `i`th expression. For constraints, it also needs to return the inequality form `f(x) <= rhs`, instead of just the body `f(x)`.